cpp/tests/lists/contains_tests.cpp

```diff
@@ -0,0 +1,631 @@
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+struct ContainsTest : public cudf::test::BaseFixture {
+};
+
+using ContainsTestTypes = cudf::test::
+  Concat<cudf::test::IntegralTypesNotBool, cudf::test::FloatingPointTypes, cudf::test::ChronoTypes>;
+
+template <typename T>
+struct TypedContainsTest : public ContainsTest {
+};
+
+TYPED_TEST_CASE(TypedContainsTest, ContainsTestTypes);
+
+namespace {
+template <typename T, std::enable_if_t<cudf::is_numeric<T>(), void>* = nullptr>
+auto create_scalar_search_key(T const& value)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_numeric_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(true);
+  static_cast<scalar_type_t<T>*>(search_key.get())->set_value(value);
+  return search_key;
+}
+
+template <typename T, std::enable_if_t<std::is_same<T, std::string>::value, void>* = nullptr>
+auto create_scalar_search_key(std::string const& value)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  return make_string_scalar(value);
+}
+
+template <typename T, std::enable_if_t<cudf::is_timestamp<T>(), void>* = nullptr>
+auto create_scalar_search_key(typename T::rep const& value)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_timestamp_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(true);
+  static_cast<scalar_type_t<typename T::rep>*>(search_key.get())->set_value(value);
+  return search_key;
+}
+
+template <typename T, std::enable_if_t<cudf::is_duration<T>(), void>* = nullptr>
+auto create_scalar_search_key(typename T::rep const& value)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_duration_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(true);
+  static_cast<scalar_type_t<typename T::rep>*>(search_key.get())->set_value(value);
+  return search_key;
+}
+
+template <typename T, std::enable_if_t<cudf::is_numeric<T>(), void>* = nullptr>
+auto create_null_search_key()
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_numeric_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(false);
+  return search_key;
+}
+
+template <typename T, std::enable_if_t<cudf::is_timestamp<T>(), void>* = nullptr>
+auto create_null_search_key()
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_timestamp_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(false);
+  return search_key;
+}
+
+template <typename T, std::enable_if_t<cudf::is_duration<T>(), void>* = nullptr>
+auto create_null_search_key()
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  auto search_key = make_duration_scalar(data_type{type_to_id<T>()});
+  search_key->set_valid(false);
+  return search_key;
+}
+
+}  // namespace
+
+TYPED_TEST(TypedContainsTest, ListContainsScalarWithNoNulls)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
```

Comment:
These lines present in every TEST and every function. Move it to file level (as long as it doesn't cause collision).

Reply:
I don't think a global `using namespace` statement is a good idea. However, there are some places where this could be eliminated altogether, like this one:


```template <typename T, std::enable_if_t<std::is_same<T, std::string>::value, void>* = nullptr>
auto create_scalar_search_key(std::string const& value)
{
  using namespace cudf;
  using namespace cudf::test;

  return make_string_scalar(value);
}
```

Just change it to 

```
template <typename T, std::enable_if_t<std::is_same<T, std::string>::value, void>* = nullptr>
auto create_scalar_search_key(std::string const& value)
{
  return cudf::make_string_scalar(value);
}
```

Reply:
another alternative common pattern in our tests is to put all the tests inside `namespace cudf {  namespace test {`
is that a better idea or worse?

Reply:
Argh, I missed this discussion and committed the global `using namespace` change. Given the advice, I figured this was acceptable in CUDF. 

I'll revert this.

Reply:
> put all the tests inside `namespace cudf { namespace test {`

That does sound better.